### DISCUSSION TITLE
Connection problem ECU R PRO

### DISCUSSION THREAD

**Discussion by Felixlechatt** [ksheumaker/homeassistant-apsystems_ecur#150] [General]
Log Home assistant v1.2.26 version:
  File "/config/custom_components/apsystems_ecur/APSystemsSocket.py", line 107, in query_ecu
    self.open_socket()
  File "/config/custom_components/apsystems_ecur/APSystemsSocket.py", line 104, in open_socket
    raise APSystemsInvalidData(err)
custom_components.apsystems_ecur.APSystemsSocket.APSystemsInvalidData: [Errno 111] Connection refused

**Comment by HAEdwin** [comment]
Tell me more... is this a new install? Have you used the integration before and did it work then? Did you read the manual an have you tried NC [IP-address] ?

**Reply by Felixlechatt** [reply]
this is a new install, i have delete and ré-install apps. i've never succeful connection.
i've install "terminal & ssh" to Home assistant for:
NC 192.168.1.50 8899
but any response to command 

with my pc to W+R -> cmd -> ping IP (OK) i've a response 

I don't undestand why is not Ok

PS: i'm french sorry for i don't speak very well :)

**Reply by HAEdwin** [reply]
Make absolutely sure you have the right node and the ECU is WiFi enabled (ethernet disconnected). Ping to the ECU with -t and power down the ECU to see if you have the right node. Assign a fixed IP-Address to the ECU using the MAC address of the WiFi interface. Because the ECU is new as well comment line 116 to 119 in APSystemsSocket.py for a while until lifetime energy on the ECU is not zero anymore.
![image](https://user-images.githubusercontent.com/82239374/215358265-2ab47d01-3c1a-4e59-b3c2-278941fcabdd.png)

**Reply by Felixlechatt** [reply]
I connected my Ecu-R to Ethernet and not Wifi, i modified to wifi and disabled connection ethernet.
I surprised to Name of ecu-R Wifi "ESP 0856" strange for me because the IP WIFI is not egal To ethernet.
Now, this Ecu-R succefull connect with Home assistant but after few hour later => disconnect auto (no signal or signal loose) the solution is reboot Ecu-R for new signal zigbee connection.

![image](https://user-images.githubusercontent.com/79603807/216129207-c2d29d86-5d4a-43e1-92d5-a58f1427ea7c.png)
exemple with this image

**Reply by HAEdwin** [reply]
Try to move the ECU to a better position for Zigbee as well as WiFi.

**Comment by HAEdwin** [comment]
I'll close this discussion, might any other issue arise please create a new ticket.
